dlt/extract/incremental.py

```diff
@@ -0,0 +1,133 @@
+from hashlib import sha1
+from typing import Generic, TypeVar, Any, Optional, Callable, Protocol, List, TypedDict, cast, Union, Tuple, get_origin, Dict
+import inspect
+
+from functools import wraps
+
+from dlt.common.json import json
+from dlt.common.typing import TDataItem, AnyFun, TFun
+from dlt.common.configuration import configspec, known_sections, with_config, resolve_configuration
+from dlt.common.configuration.specs import BaseConfiguration
+from dlt.common.configuration.specs.exceptions import NativeValueError
+from dlt.common.configuration.accessors import config
+
+
+TCursorValue = TypeVar("TCursorValue", bound=Any)
+TUniqueValue = TypeVar("TUniqueValue", bound=Any)
+LastValueFunc = Callable[..., Any]
+
+
+class IncrementalColumnState(TypedDict):
+    last_value: Optional[Any]
+    unique_keys: List[Any]
+
+
+@configspec
```

Comment:
why `configspec`? do you want to make this part configurable and specify it via `config.toml`? will it work separately for each resource?